### ISSUE TITLE
.NFO files?

### ISSUE THREAD

**Issue by tharubberchicken** [rmcrackan/Libation#126]
Not really a bug, but I'm _sure_ there used to be .NFO files downloaded with each book that had some useful info like bit/sample rate and the book description. In earlier versions at least, unless I missed something and this feature was removed. Or it is supposed to be working, but it is not for me. Then it perhaps _is_ a bug. 😁

**Comment by rmcrackan** [comment]
tl;dr -- choice, not bug. Removed in one of the most recent updates. Is it something you used regularly?

long version: as Libation offered more options this became harder to maintain for something that it didn't seem like was used much. We used to download the book file then create the nfo next to it. Since then, audible has changed their file type, we've started decrypting directly during download (instead of download the whole file then decrypt), we offered download + decrypt directly to mp3, we added convert m4b => mp3 after the fact, and recently audible started offering non-encrypted files. Each of these has to go through a different part of the program and pushing the data around which is needed for the nfo file became death by 1,000 cuts.

**Comment by tharubberchicken** [comment]
That's okay, it's not a deal breaker for me. I can do without it. 👍🏻 There seems to be a limit to the number of characters an M4A/M4B comment tag can have and it often truncated longer book descriptions. That was really the only reason I used them.

**Comment by wtanksleyjr** [comment]
The "export library" feature still works, right? That gets the data way
better than what the tags have.

-Wm

**Comment by Mbucari** [comment]
> There seems to be a limit to the number of characters an M4A/M4B comment tag can have and it often truncated longer book descriptions. That was really the only reason I used them.

There are two metadata boxes containing description information:  ©cmt and ©des. ©cmt Contains the description truncated to 200-some characters. That is also the box that most media players will read and display.  ©des Contains the full-length description, but many media players do not display its contents. I assure you, the long description inside ©des is what used to be inside the NFO, and is still inside the M4B files 

The full description should be in the

**Comment by flysalad** [comment]
I like to have the .nfo file as this provides the full details for the book and helps a lot when you have 1,000+ books. I particularly like the full-length description which is missing from the exported Libation data. I have to get it directly from the SQLite db.

**Comment by tharubberchicken** [comment]
> > There seems to be a limit to the number of characters an M4A/M4B comment tag can have and it often truncated longer book descriptions. That was really the only reason I used them.
> 
> There are two metadata boxes containing description information: ©cmt and ©des. ©cmt Contains the description truncated to 200-some characters. That is also the box that most media players will read and display. ©des Contains the full-length description, but many media players do not display its contents. I assure you, the long description inside ©des is what used to be inside the NFO, and it is still inside the M4B files.

Actually, this is good enough for me. As long as the complete book description is contained _somewhere_ in the audio file then I am happy. I did not know about this second _tag_ and will investigate when I get back to a PC. Thanks!

**Comment by rmcrackan** [comment]
> The "export library" feature still works, right? That gets the data way better than what the tags have.

@wtanksleyjr -- yes, they are unchanged. The nfo files also contained file specific info, like what's in the metadata tags and audio bit-rate.

@flysalad

> I like to have the .nfo file as this provides the full details for the book and helps a lot when you have 1,000+ books. I particularly like the full-length description which is missing from the exported Libation data. I have to get it directly from the SQLite db.

Have you tried the export feature? It will export to json, csv, or excel and is available both in the GUI and CLI so you can even script it. I encourage the export feature rather than the db. You're welcome to poke around the db (it's your data after all) but a database is not an api. I can and will make radical changes to the db and I make no assurance that any other program that uses it directly will continue to work. This is why I made the exports.
